source/scripts/user.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * This file contains functions to create a user account, store the user account to local storage,
+ * login to a user account that help manipulate the account system. A User class might not be needed
+ * to create a general object for a user to be stored in local storage.
+ * -------------------------------------
+ * Some stretch goals:
+ * - delete an account and delete all the posts on that account (?)
+ * - retype password to confirm when creating an account
+ * - "Forgot password" option in login page
+ * - when someone enters the page containg the domain of our app, redirect to login page if not logged in
+ */
+
+// constant urls
+const APP_MAIN_URL = "some_url_for_the_main_app_page";
+const CREATE_ACCOUNT_URL = "some_url_for_user_to_create_account";
+const LOGIN_URL = "some_url_for_user_to_login_in";
+
+/**
+ * Class for creating User type Object, could instead just declare a general Object type without this
+ */
+class User extends HTMLElement() {
+    constructor(){
+        super();
+        // to add something that needed to be displayed on html page
+    }
+    set data(data){
+        // data that need to be visualized on html page
+    }
+}
+
+// customElements.define('user-account', User); // custom element from User Object, might not be needed
+
+/**
+ * Get the users from local storage as an array of user objects
+ */
+function get_users_from_storage() {
+    if (!localStorage.getItem('users')) {
+      let empty_array = [];
+      return empty_array;
+    }
+    return JSON.parse(localStorage.getItem('users'));

```

Comment:
the 'users' string should probably be changed to a constant defined at the top of the file